src/realm/cluster_tree.cpp

```diff
@@ -577,8 +596,8 @@ bool ClusterNodeInner::get_leaf(ObjKey key, ClusterNode::IteratorState& state) c
 
     size_t sz = node_size();
     while (child_ndx < sz) {
-        int64_t key_offset = m_keys.is_attached() ? m_keys.get(child_ndx) : (child_ndx << m_shift_factor);
-        ObjKey new_key(key_offset < key.value ? key.value - key_offset : 0);
+        auto key_offset = m_keys.is_attached() ? m_keys.get(child_ndx) : (child_ndx << m_shift_factor);
```

Comment:
I'm unclear what type key_offset is supposed to be here. If it's `m_keys.get()` then it's uint64_t, but if its `child_ndx << m_shift_factor` then it's size_t. On most platforms that's a uint64_t, but it's broadly undefined. Can we pick one here and use that instead of auto.

Reply:
Sure